bitnami/spark/README.md

```diff
@@ -0,0 +1,256 @@
+| `security.ssl.enabled`                      | Enable the SSL configuration                                                                | `no`                                                    |
+| `security.ssl.needClientAuth`               | Enable the client authentication                                                            | `no`                                                    |
+| `security.ssl.protocol`                     | Set the SSL protocol                                                                        | `TLSv1.2`                                               |
+| `security.certificatesSecretName`           | Set the name of the secret that contains the certificates                                   | No default                                              |
+| `service.type`                              | Kubernetes Service type                                                                     | `ClusterIP`                                             |
+| `service.port`                              | spark client port                                                                           | `2379`                                                  |
+| `service.nodePort`                          | Port to bind to for NodePort service type (client port)                                     | `nil`                                                   |
+| `service.annotations`                       | Annotations for spark service                                                               | {}                                                      |
+| `service.loadBalancerIP`                    | loadBalancerIP if spark service type is `LoadBalancer`                                      | `nil`                                                   |
+| `ingress.enabled`                           | Enable the use of the ingress controller to access the web UI                               | `true`                                                  |
+| `ingress.hosts`                             | Add hosts to the ingress controller with name and path                                      | `name: spark.local`, `path: /`                          |
+
+
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install --name my-release \
+  --set master.webPort=8081 bitnami/spark
+```
+
+The above command sets the spark master web port to `8081`.
+
+Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,
+
+```console
+$ helm install --name my-release -f values.yaml bitnami/spark
+```
+
+> **Tip**: You can use the default [values.yaml](values.yaml)
+
+### Production configuration and horizontal scaling
```

Comment:
What order did you follow to list the parameters? In other charts we follow the same order as the parameters appear in the files. Also, can you ensure that running a diff between both files only the parameters mentioned here appears? Check `customClusterPort`
```
$ colordiff value value-prod
103a104
>   customClusterPort: false
190c191
<   # passwordsSecretName:
---
>   passwordsSecretName: my-passwords-secret
194,195c195,196
<     authenticationEnabled: no
<     encryptionEnabled: no
---
>     authenticationEnabled: yes
>     encryptionEnabled: yes
198c199
<   storageEncryptionEnabled: no
---
>   storageEncryptionEnabled: yes
202,203c203,204
<     enabled: no
<     needClientAuth: no
---
>     enabled: yes
>     needClientAuth: yes
208c209
<   # certificatesSecretName:
---
>   certificatesSecretName: my-certificates-secret
```

Reply:
```➜  spark git:(sparkChart) ✗ diff values-production.yaml values.yaml
31a32,35
>
>   ## Set a custom configuration by using an existing configMap with the configuration file.
>   # configurationConfigMap
>
105a110,112
>   ## Set a custom configuration by using an existing configMap with the configuration file.
>   # configurationConfigMap:
>
147a155
>
203,204c211,212
<     authenticationEnabled: yes
<     encryptionEnabled: yes
---
>     authenticationEnabled: no
>     encryptionEnabled: no
207c215
<   storageEncryptionEnabled: yes
---
>   storageEncryptionEnabled: no
211,212c219,220
<     enabled: yes
<     needClientAuth: yes
---
>     enabled: no
>     needClientAuth: no```

Reply:
??